core/trino-main/src/main/java/io/trino/execution/SqlQueryExecution.java

```diff
@@ -378,6 +382,12 @@ public void start()
                     // query already started or finished
                     return;
                 }
+                Thread planningThread = currentThread();
+                stateMachine.getStateChange(PLANNING).addListener(() -> {
+                    if (stateMachine.getQueryState() == FAILED) {
+                        planningThread.interrupt();
```

Comment:
This might not be a planning thread anymore (when planning is finished) when query goes to `FAILED` state. In fact, this executor thread might be running planning for some other query. I think interruption needs to run in critical section:
1. planning thread is setting `planningThread=null` when planning is finished
2. other thread is calling `interrupt`

Reply:
The listener will fire only when the state changes from `PLANNING` to `FAILED`. Once planning is complete the state changes immediately to `STARTING` and the thread will not get interrupted.

Reply:
Ok, you are right, the thread might be reused. I will sync it.

Reply:
Fixed. PTAL